### PR TITLE
Expire projection and filter cache entries after one hour

### PR DESCRIPTION
**Pull request by wenleix**
We have observed deoptimization storm that leads to slowness.
We suspect that it is a JVM bug that is related to stale/corrupted
profiling data associated with generated classes.

### CODE REVIEW

**Comment by wenleix** [comment]
cc @AlekseiS , @jessesleeping , @hellium01

**Comment by wenleix** [comment]
This has some resemblance with https://github.com/prestodb/presto/pull/9189

File: presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java [hunk 1]
```diff
@@ -119,10 +120,15 @@ public PageFunctionCompiler(Metadata metadata, int expressionCacheSize)
         this.metadata = requireNonNull(metadata, "metadata is null");
         this.determinismEvaluator = new DeterminismEvaluator(metadata.getFunctionRegistry());
 
+        // We have observed deoptimization storm that leads to slowness.
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java]: Grammar nit
```java
// We have observed deoptimization storms that lead to slowness.

**Comment by wenleix** [comment]
See the discussion in JDK mailing list: http://mail.openjdk.java.net/pipermail/hotspot-dev/2017-September/028423.html

The phenomenon is that in jstack, we see many threads are trying to compile the exact
same lambda form. Example stacktrace:
```
    "20170926_232912_39740_3vuuu.1.79-4-76640" #76640 prio=5 os_prio=0
tid=0x00007f908006dbd0 nid=0x150a6 runnable [0x00007f8bddb1b000]
       java.lang.Thread.State: RUNNABLE
            at sun.misc.Unsafe.defineAnonymousClass(Native Method)
            at java.lang.invoke.InvokerBytecodeGenerator.
loadAndInitializeInvokerClass(InvokerBytecodeGenerator.java:284)
            at java.lang.invoke.InvokerBytecodeGenerator.loadMethod(
InvokerBytecodeGenerator.java:276)
            at java.lang.invoke.InvokerBytecodeGenerator.
generateCustomizedCode(InvokerBytecodeGenerator.java:618)
            at java.lang.invoke.LambdaForm.compileToBytecode(LambdaForm.
java:654)
            at java.lang.invoke.LambdaForm.prepare(LambdaForm.java:635)
            at java.lang.invoke.MethodHandle.updateForm(MethodHandle.java:
1432)
            at java.lang.invoke.MethodHandle.customize(MethodHandle.java:
1442)
            at java.lang.invoke.Invokers.maybeCustomize(Invokers.java:407)
            at java.lang.invoke.Invokers.checkCustomized(Invokers.java:398)
            at java.lang.invoke.LambdaForm$MH/170067652.invokeExact_MT(
LambdaForm$MH)
            at com.facebook.presto.operator.aggregation.MinMaxHelper.
combineStateWithState(MinMaxHelper.java:141)
            at com.facebook.presto.operator.aggregation.
MaxAggregationFunction.combine(MaxAggregationFunction.java:108)
            at java.lang.invoke.LambdaForm$DMH/1607453282.invokeStatic_
L3_V(LambdaForm$DMH)
            at java.lang.invoke.LambdaForm$BMH/1118134445.reinvoke(
LambdaForm$BMH)
            at java.lang.invoke.LambdaForm$MH/1971758264.
linkToTargetMethod(LambdaForm$MH)
            at com.facebook.presto.$gen.IntegerIntegerMaxGroupedAccumu
lator_3439.addIntermediate(Unknown Source)
            at com.facebook.presto.operator.aggregation.builder.
InMemoryHashAggregationBuilder$Aggregator.processPage(
InMemoryHashAggregationBuilder.java:367)
            at com.facebook.presto.operator.aggregation.builder.
InMemoryHashAggregationBuilder.processPage(InMemoryHashAggregationBuilder
.java:138)
            at com.facebook.presto.operator.HashAggregationOperator.
addInput(HashAggregationOperator.java:400)
            at com.facebook.presto.operator.Driver.processInternal(Driver.
java:343)
            at com.facebook.presto.operator.Driver.lambda$processFor$6(
Driver.java:241)
            at com.facebook.presto.operator.Driver$$Lambda$765/442308692.get(Unknown
Source)
            at com.facebook.presto.operator.Driver.tryWithLock(Driver.
java:614)
            at com.facebook.presto.operator.Driver.processFor(Driver.java:
235)
            at com.facebook.presto.execution.SqlTaskExecution$
DriverSplitRunner.processFor(SqlTaskExecution.java:622)
            at com.facebook.presto.execution.executor.
PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
            at com.facebook.presto.execution.executor.TaskExecutor$
TaskRunner.run(TaskExecutor.java:485)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(
ThreadPoolExecutor.java:1142)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(
ThreadPoolExecutor.java:617)
            at java.lang.Thread.run(Thread.java:748)
    ...
```

**Pull request merged**